src/app/game/game-form/game-form.component.ts

```diff
@@ -1,24 +1,65 @@
-import { Component, HostBinding } from "@angular/core";
+import { Component, HostBinding, OnDestroy, OnInit } from "@angular/core";
 import { FormControl, Validators } from "@angular/forms";
 import { GameService } from "../game.service";
+import { ActivatedRoute, Router } from "@angular/router";
+import { Subscription, filter, map } from "rxjs";
+import { GameT } from "../game.types";
+
+type ModeT = "Adicionar" | "Atualizar";
 
 @Component({
   selector: "app-game-form",
   templateUrl: "./game-form.component.html",
 })
-export class GameFormComponent {
+export class GameFormComponent implements OnInit, OnDestroy {
   @HostBinding("class") classes = "flex flex-col gap-2";
 
   name = new FormControl("", [Validators.required, Validators.minLength(3)]);
+  private gameId!: string;
+
+  mode: ModeT = "Adicionar";
+  private routeSubscription!: Subscription;
+
+  constructor(
+    private gameService: GameService,
+    private activatedRoute: ActivatedRoute,
+    private router: Router,
+  ) {}
+
+  ngOnInit(): void {
+    this.routeSubscription = this.activatedRoute.paramMap
+      .pipe(
+        map(() => window.history.state),
+        filter((game) => !!game.id),
+      )
+      .subscribe((game: GameT) => {
+        this.name.setValue(game.name);
+        this.gameId = game.id;
+        this.mode = "Atualizar";
+      });
+  }
 
-  constructor(private gameService: GameService) {}
+  ngOnDestroy(): void {
+    this.routeSubscription.unsubscribe();
+  }
 
-  public submitForm(): void {
+  submitForm(): void {
     if (this.name.valid) {
-      this.gameService.newGame(this.name.value as string).subscribe();
+      if (this.mode === "Adicionar") {
+        this.gameService.newGame(this.name.value as string).subscribe();
+      } else {
+        this.gameService
+          .updateGame(this.gameId, this.name.value as string)
+          .subscribe();
+      }
       this.name.reset();
+      this.navigateToList();
```

Comment:
a navegação para a lista acontece independente do retorno da requisição ao backend ter sido bem sucedida ou não, uma solução é chamar `this.navigateToList()` nos callbacks dos `subscribe`s.